File: Common/tree_gen.m
function tree_gen(tree_root,c)
% generates a directory structure using the time vector from clock
% tree_root is where you want the subdirectoires to be placed
% cl is the clock vector

% Adapting to windows or linux paths
if ispc == 1
    slash = '\';
else
    slash = '/';
end

yr = num2str(c(1));
mth = num2str(c(2));
dy = num2str(c(3));

if c(2) < 10
mth = ['0' mth];
end
if c(3) < 10
dy = ['0' dy];
end

% generating directory tree
if exist([tree_root yr slash],'dir') == 0
    system(['mkdir ' tree_root  yr slash]);
end
if exist([tree_root yr slash mth slash],'dir') == 0
    system(['mkdir ' tree_root yr slash mth '/']);
end
if exist([tree_root yr slash mth slash dy slash],'dir') == 0
    system(['mkdir ' tree_root yr slash mth slash dy slash]);
end

File: python_port/Common/tree_gen.py
import os


def treeGen(tree_root, c):
    """Generates a directory structure using the time vector from clock
    tree_root is where you want the subdirectoires to be placed
    cl is the clock vector
    """
    yr = c[0]
    mth = c[1]
    dy = c[2]

    if mth < 10:
        mth = "".join(("0", mth))

    if dy < 10:
        dy = "".join(("0", dy))

#    if not os.path.isdir(os.path.join(tree_root, yr)):
#        os.mkdir(os.path.join(tree_root, yr))

#    if not os.path.isdir(os.path.join(tree_root, yr, mth)):
#        os.mkdir(os.path.join(tree_root, yr, mth))

    if not os.path.isdir(os.path.join(tree_root, yr, mth, dy)):
        os.mkdirs(os.path.join((tree_root, yr, mth, dy)))
